confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/internal/input/ConfluenceSpaceHelpers.java

```diff
@@ -0,0 +1,195 @@
+     * @since 9.88.5
+     */
+    public void setProperties(ConfluenceInputProperties properties)
+    {
+        this.properties.set(properties);
+    }
+
+    /**
+     * Checks whether a given space is forbidden to be overwritten based on the configured forbidden spaces.
+     * @param spaceKey the spaceKey of the space that is checked
+     * @return true if the specified space is listed among the forbidden spaces and cannot be overwritten
+     * @since 9.88.5
+     */
+    public boolean checkIfTheSpaceOverwriteIsForbidden(String spaceKey)
+    {
+        Set<String> forbiddenSpaces = this.properties.get().getForbiddenSpaces();
+
+        if (forbiddenSpaces == null || forbiddenSpaces.isEmpty()) {
+            return false;
+        }
+
+        EntityReference rootReference = this.properties.get().getRoot();
+
+        SpaceReference spaceReference = getRootWithWikiSpaceReference(rootReference, spaceKey);
+
+        for (String forbiddenSpace : forbiddenSpaces) {
+            EntityReference forbiddenRef =
+                entityReferenceResolver.resolve(forbiddenSpace, EntityType.SPACE);
+            if (forbiddenRef.equals(spaceReference)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /**
+     * @param spaceKey the spaceKey of the space that is checked
+     * @param checkIfConfluenceImported flag that specifies whether to check if another space imported from
+     *     Confluence already exists, or to simply search for any space.
+     * @return a user reference.
+     * @since 9.88.5
+     */
+    public boolean checkIfSpaceExists(String spaceKey, Boolean checkIfConfluenceImported)
+    {
+        EntityReference rootReference = this.properties.get().getRoot();
+
+        SpaceReference spaceReference = getRootWithWikiSpaceReference(rootReference, spaceKey);
+
+        String spaceName = spaceReference.getName();
+
+        String queryString =
+            "select count(doc) from XWikiDocument as doc"
+                + " where doc.space = :spaceName or doc.space like :spacePrefix";
+
+        try {
+            List<Long> result = queryManager.createQuery(queryString, Query.HQL)
+                .setWiki(spaceReference.getWikiReference().getName())
+                .bindValue("spaceName", spaceName)
+                .bindValue("spacePrefix", spaceName + ".%")
+                .execute();
+
+            if (result.isEmpty() || result.get(0) == 0) {
+                return false;
+            }
+
+            if (checkIfConfluenceImported) {
+                String webHomeString = "WebHome";
+                XWikiContext xContext = contextProvider.get();
+                DocumentReference docRef;
+
+                if (rootReference != null) {
+                    docRef = new DocumentReference(webHomeString, spaceReference);
+                } else {
+                    docRef = new DocumentReference(xContext.getWikiId(), spaceKey, webHomeString);
+                }
```

Comment:
If you apply the earlier suggestion, this becomes:

```suggestion
                DocumentReference docRef = new DocumentReference("WebHome", rootReference);
```